extension/src/popup/components/account/NotFundedMessage.tsx

```diff
@@ -40,27 +46,61 @@ const NotFundedHeaderEl = styled.h3`
   }
 `;
 
-export const NotFundedMessage = () => (
-  <NotFundedWrapperEl>
-    <NotFundedHeaderEl>This Stellar address is not funded</NotFundedHeaderEl>
-    <p>To create this account, fund it with a minimum of 1 XLM.</p>
-    <p>
-      <a
-        href="https://developers.stellar.org/docs/tutorials/create-account/#create-account"
-        rel="noreferrer"
-        target="_blank"
-      >
-        Learn more about account creation
-      </a>
-    </p>
-    <p>
-      <a
-        href="https://www.stellar.org/lumens/exchanges"
-        rel="noreferrer"
-        target="_blank"
-      >
-        See where you can buy lumens
-      </a>
-    </p>
-  </NotFundedWrapperEl>
-);
+export const NotFundedMessage = ({
+  isTestnet,
+  publicKey,
+  setIsAccountFriendbotFunded,
+}: {
+  isTestnet: boolean;
+  publicKey: string;
+  setIsAccountFriendbotFunded: (isAccountFriendbotFunded: boolean) => void;
+}) => {
+  const dispatch = useDispatch();
+
+  const handleFundAccount = async () => {
+    await dispatch(fundAccount(publicKey));
+    setIsAccountFriendbotFunded(true);
```

Comment:
Could we run into issues with timing if we trigger fetching of the account info before funding is done?

Reply:
That's a good question. `fundAccount()` doesn't resolve until the funding request has completed, so `setIsAccountFriendbotFunded` is only triggered once we're done. In a Slow 3G test, for ex, the UI stays in a loading state while we're making the funding request and then finally re-fetches once that promise has resolved